tests/unittests/tests-libcose/tests-libcose.c

```diff
@@ -0,0 +1,222 @@
+﻿/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ * Copyright (C) 2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+  * @ingroup    tests
+  * @{
+  *
+  * @file
+  * @brief      Unit tests for pkg libcose
+  *
+  * @author     Koen Zandberg <koen@bergzand.net>
+  */
+
+#define MAX_NUMBER_BLOCKS 128
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <tweetnacl.h>
+#include "cn-cbor/cn-cbor.h"
+#include "cose.h"
+#include "cose/crypto.h"
+#include "memarray.h"
+#include "embUnit.h"
+
+#include "tests-libcose.h"
+
+static char kid[] = "koen@riot-os.org";
+static char kid2[] = "paco@riot-os.org";
+static unsigned char pk[crypto_sign_PUBLICKEYBYTES];
+static unsigned char sk[crypto_sign_SECRETKEYBYTES];
+static unsigned char pk2[crypto_sign_PUBLICKEYBYTES];
+static unsigned char sk2[crypto_sign_SECRETKEYBYTES];
+
+#define MLEN (sizeof(payload))
+#define SMLEN (sizeof(payload)+ crypto_sign_BYTES)
+#define NUM_TESTS (sizeof(tests)/sizeof(struct test))
+/* COSE sign buffer */
+static uint8_t buf[512];
+/*Signature Verification buffer */
+static uint8_t vbuf[256];
+
+
+static cn_cbor block_storage_data[MAX_NUMBER_BLOCKS];
+static memarray_t storage;
+static int cur_alloc = 0;
+static int max_alloc = 0;
+
+static void setUp(void)
+{
+    /* Initialize */
+    random_init(0);
+}
+
+static void tearDown(void)
+{
+    /* Finalize */
+}
+
+static void print_bytestr(uint8_t *bytes, size_t len)
+{
+    for(unsigned int idx=0; idx < len; idx++)
+    {
+        printf("%02X", bytes[idx]);
+    }
+}
+
+
+/* CN_CBOR calloc functions */
+static void *cose_calloc(size_t count, size_t size, void *memblock)
+{
+    (void)count;
+    (void)size;
+    void *block = memarray_alloc(memblock);
+    if (block) {
+        memset(block, 0, sizeof(cn_cbor));
+        cur_alloc++;
+        if (cur_alloc > max_alloc)
+        {
+            max_alloc = cur_alloc;
+        }
+    }
+    return block;
+
+}
+
+static void cose_free(void *ptr, void *memblock)
+{
+    cur_alloc--;
+    memarray_free(memblock, ptr);
+}
+
+static cn_cbor_context ct =
+{
+    .calloc_func = cose_calloc,
+    .free_func = cose_free,
+    .context = &storage,
+};
```

Comment:
I'd advice to either put it at the beginning (with indentation) where you declare all the global variables, or you add some indentation to identify it better. At a first glance it looks like a malformed function.

Reply:
Changed it a bit, can you check if it looks less like a malformed function now?

Reply:
It's OK now.